### DISCUSSION TITLE
Fix footnote styles to work with latest Sphinx + docutils

### DISCUSSION THREAD

**Discussion by jab** [pradyunsg/furo#489] [Potential Issue]
### What's happening?

A new version of Sphinx has now been released, which changes its dependency pins to allow docutils 0.19, which includes a fix for [New HTML markup for footnotes is difficult to stylise](https://sourceforge.net/p/docutils/bugs/450/), along with corresponding [Sphinx changes](https://github.com/sphinx-doc/sphinx/pull/10599).

Now that both Sphinx and docutils have made stable releases to address this issue, I believe Furo is no longer blocked from writing styles that fix the regression in footnote alignment that was caused by the changes in docutils 0.18. (The latest version of Furo is still causing the broken looking footnotes that are visible in [this screenshot](https://github.com/pradyunsg/furo/discussions/452#discussion-4121227) when used with latest Sphinx (5.1.1) and docutils (0.19).)

### Reproducer

1. Build Sphinx documentation at https://github.com/jab/bidict by running `pip install -r requirements/docs.txt && make -C docs html` from a fresh clone.
2. Open `docs/_build/html/index.html` in a browser.
3. Scroll down to the footnotes at the bottom of the page.
4. Observe that (1) footnotes are no longer appearing on the same line as their corresponding footnote numbers, and (2), they're not even being indented consistently: Note how footnote 3 is not indented, while footnotes 1, 2, and 4 are indented:

![](https://user-images.githubusercontent.com/64992/172055241-ecb1a308-6e98-4274-ab91-482d58fb1a35.png)

Ref: https://github.com/pradyunsg/furo/discussions/452

### Expectation

Alignment of footnotes stays consistent and correct, as it appeared before the changes introduced in docutils 0.18 caused this breakage:

![](https://user-images.githubusercontent.com/64992/172055197-581e753d-469f-4297-b490-2c39f4e9b0c7.png)

### Code of Conduct

- [X] I agree to follow the [Code of Conduct](https://github.com/pradyunsg/furo/blob/main/CODE_OF_CONDUCT.md).

**Comment by jab** [comment]
Hi @pradyunsg, I see you converted the [issue created for this](https://github.com/pradyunsg/furo/issues/488) into this discussion.

Just wanted to explain that I created an issue rather than a discussion because we already had a prior discussion about this in https://github.com/pradyunsg/furo/discussions/452, and I interpreted your much-appreciated response there and your followup reporting to the Sphinx and docutils projects as acknowledgment that you would welcome a trackable issue for this, now that there is an unblocked piece of work that would fix this in Furo, when used with recent enough Sphinx + docutils. I'm sorry I misunderstood, and I hope you don't mind.

If there are any other thoughts you can share to clarify this issue, I'd be grateful to understand better. Thanks again for all your great work on Furo and in the community in general.

**Comment by jab** [comment]
@pradyunsg, I followed https://docs.readthedocs.io/en/stable/guides/adding-custom-css.html#adding-custom-css-or-javascript-to-sphinx-documentation to work around this in a custom.css file. If my workaround generalizes, then the following patch to Furo would fix this:
```diff
diff --git a/src/furo/assets/styles/content/_footnotes.sass b/src/furo/assets/styles/content/_footnotes.sass
index 608f895d..4afdd721 100644
--- a/src/furo/assets/styles/content/_footnotes.sass
+++ b/src/furo/assets/styles/content/_footnotes.sass
@@ -40,4 +40,5 @@ div.citation > span

 aside.footnote > p,
 div.citation > p
-  margin-left: 2rem
+  margin-left: 1rem
+  display: inline-block
```
Would you be willing to give this two-line change a try with one of your projects that uses footnotes, and if all looks good to you, apply the changes?

**Reply by pradyunsg** [reply]
Hi there! I’m pretty sure that breaks multi-line footnotes.

**Reply by jab** [reply]
Thanks for getting back to me on this!

I tried make my window narrower so that one of my footnotes becomes multi-line, and this is still strictly an improvement over the current styles:

<details><summary>Before narrowing:</summary>
<img width="734" alt="Screen Shot 2022-08-06 at 4 08 37 PM" src="https://user-images.githubusercontent.com/64992/183264681-66814db2-edd8-4a93-99e2-7cdc8328fc0f.png">
</details>
<details><summary>After narrowing:</summary>
<img width="556" alt="Screen Shot 2022-08-06 at 4 08 47 PM" src="https://user-images.githubusercontent.com/64992/183264727-a097e129-5494-4cff-91a5-4aa0dbe14603.png">
</details>

Compare these to the first screenshot in my [initial post in this discussion](https://github.com/pradyunsg/furo/discussions/489#discussion-4254835) and I think you'll see what I mean.

Would you be willing to accept these changes as an incremental improvement?

**Reply by pradyunsg** [reply]
Try making that change, and looking at the footnotes section in kitchen-sink/generic/ -- the multi-paragraph footnotes will render in a suboptimal manner.

**Reply by jab** [reply]
Ah, with a multi-paragraph footnote (not just multi-line) I see what you mean, thanks.

Here's another approach:
```diff
diff --git a/src/furo/assets/styles/content/_footnotes.sass b/src/furo/assets/styles/content/_footnotes.sass
index 608f895d..028e0499 100644
--- a/src/furo/assets/styles/content/_footnotes.sass
+++ b/src/furo/assets/styles/content/_footnotes.sass
@@ -36,7 +36,9 @@ aside.footnote
 
 aside.footnote > span,
 div.citation > span
+  float: left
   font-weight: 500
+  padding-right: 0.25rem
 
 aside.footnote > p,
 div.citation > p
```
And here's what the kitchen sink footnotes look like with this applied:
<img width="686" alt="Screen Shot 2022-08-07 at 8 12 51 PM" src="https://user-images.githubusercontent.com/64992/183316734-65a7ca17-bf6f-4e63-93c0-bcce42e147d5.png">

What do you think?

**Reply by pradyunsg** [reply]
You know what, sure -- could you file a PR for this?

**Reply by jab** [reply]
Sure, submitted https://github.com/pradyunsg/furo/pull/576. PTAL and thanks for your consideration.
